beacon_node/eth2-libp2p/src/rpc/handler.rs

```diff
@@ -345,6 +376,12 @@ where
     ) {
         self.dial_negotiated -= 1;
 
+        // if shutting down, account for the recently established outbounds but drop them
+        if self.shutting_down {
+            self.update_keep_alive();
```

Comment:
Here also. We are shutting down, so tell the application that their request failed